MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -1514,6 +1514,32 @@
         _metalFeatures.minSubgroupSize = _metalFeatures.maxSubgroupSize = 4;
     }
 #endif
+	
+	bool isTimestampCounterSupported = false;
+	NSArray<id<MTLCounterSet>>* counterSets = _mtlDevice.counterSets;
+	for (id<MTLCounterSet> counterSet in counterSets) {
+		NSString* counterSetName = counterSet.name;
+		
+		if ([counterSetName caseInsensitiveCompare:MTLCommonCounterSetTimestamp] == NSOrderedSame) {
+			_mtlTimestampCounterSet = counterSet;
+			break;
+		}
+	}
+	
+	if(_mtlTimestampCounterSet != nil) {
+		NSArray<id<MTLCounter>>* countersInSet = _mtlTimestampCounterSet.counters;
+		for (id<MTLCounter> counter in countersInSet) {
+			if ([counter.name caseInsensitiveCompare:MTLCommonCounterTimestamp] == NSOrderedSame) {
+				isTimestampCounterSupported = true;
+				break;
+			}
+		}
+	}
+	
+	_metalFeatures.timestampCommandSamplingSupported = isTimestampCounterSupported &&
+		[_mtlDevice supportsCounterSampling:MTLCounterSamplingPointAtDrawBoundary] &&
+		[_mtlDevice supportsCounterSampling:MTLCounterSamplingPointAtDispatchBoundary] &&
+		[_mtlDevice supportsCounterSampling:MTLCounterSamplingPointAtBlitBoundary];
```

Comment:
The calls to `-[MTLDevice supportsCounterSampling:]` need to be in an `#if MVK_XCODE_12` block.